hypershift-operator/releaseinfo/releaseinfo.go

```diff
@@ -2,72 +2,140 @@ package releaseinfo
 
 import (
 	"bytes"
+	"context"
+	"encoding/json"
 	"fmt"
+	"io/ioutil"
 	"regexp"
 	"sort"
 	"strings"
+	"time"
 
 	"github.com/blang/semver"
 	imageapi "github.com/openshift/api/image/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/errors"
 	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/apimachinery/pkg/util/wait"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
 )
 
-type ReleaseImageInfo struct {
-	Image      string                   `json:"image"`
-	Metadata   ReleaseImageInfoMetadata `json:"metadata"`
-	References imageapi.ImageStream     `json:"references"`
+type Provider interface {
+	Lookup(image string, ctx context.Context) (*ReleaseImage, error)
 }
 
-type ReleaseImageInfoMetadata struct {
-	Kind    string `json:"kind"`
-	Version string `json:"version"`
-}
+var _ Provider = (*PodProvider)(nil)
+
+type PodProvider struct {
+	Pods v1.PodInterface
 
-type ReleaseInfo struct {
-	Images   map[string]string
-	Versions map[string]string
+	// TODO: add an ExpirationCache
 }
 
-func (info *ReleaseImageInfo) ReleaseInfo(originReleasePrefix string) (*ReleaseInfo, error) {
-	var newImagePrefix string
-	if !strings.Contains(info.Image, originReleasePrefix) {
-		newImagePrefix = strings.Replace(info.Image, ":", "-", -1)
+func (p *PodProvider) Lookup(image string, ctx context.Context) (releaseImage *ReleaseImage, err error) {
+	pod := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:    "hypershift",
+			GenerateName: "image-lookup",
+		},
+		Spec: corev1.PodSpec{
+			RestartPolicy: corev1.RestartPolicyNever,
+			Containers: []corev1.Container{
+				{
+					Name:    "lookup",
+					Image:   image,
+					Command: []string{"/usr/bin/cat", "/release-manifests/image-references"},
+				},
+			},
+		},
 	}
-	images := make(map[string]string)
-	for _, tag := range info.References.Spec.Tags {
-		name := tag.From.Name
-		if len(newImagePrefix) > 0 {
-			name = fmt.Sprintf("%s@%s", newImagePrefix, strings.Split(tag.From.Name, "@")[1])
+
+	// Launch the pod and ensure we clean up regardless of outcome
+	pod, err = p.Pods.Create(context.TODO(), pod, metav1.CreateOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to create image lookup pod: %w", err)
+	}
+	defer func() {
+		err := p.Pods.Delete(ctx, pod.Name, metav1.DeleteOptions{})
+		if err != nil {
+			err = fmt.Errorf("failed to delete image lookup pod %q: %w", pod.Name, err)
+		}
+	}()
+
+	// Wait for the pod to reach a terminate state
+	err = wait.PollImmediateUntil(1*time.Second, func() (bool, error) {
```

Comment:
nit: could be done with an informer/watch

Reply:
Perhaps... can't imagine it would be simpler and the time this entire `Lookup` method takes is negligible so I'm not inclined to change what's working yet